File: c/test2.c
#include <stdio.h>
#include <math.h>

void main(void)
{
    int a, b, c, mayor, menor, medio;

    printf("Ingrese el primer numero: ");
    scanf("%d", &a);
    printf("Ingrese el segundo numero: ");
    scanf("%d", &b);
    printf("Ingrese el tercer numero: ");
    scanf("%d", &c);

    mayor = ((2*c)+a+b+fabs(a-b)+fabs((2*c)-a-b-fabs(a-b)))/4;
    menor = ((2*c)+a+b-fabs(a-b)-fabs((2*c)-a-b+fabs(a-b)))/4;
    medio = a+b+c-mayor-menor;

    printf("\n\t%d\n", mayor);
    printf("\n\t%d\n", medio);
    printf("\n\t%d\n", menor);

    return;
}

File: python/test2.py
def main():
    print("Ordenando 3 numeros\n")

    a = int(input("Ingrese el primer numero: "))
    b = int(input("Ingrese el segundo numero: "))
    c = int(input("Ingrese el tercer numero: "))

    mayor = (((2 * a) + b + c + abs(b - c)) + abs((2 * a) - b - c - abs(b-c))) / 4
    menor = (((2 * a) + b + c - abs(b - c)) - abs((2 * a) - b - c + abs(b-c))) / 4
    medio = a + b + c - mayor - menor

    print(int(mayor))
    print(int(medio))
    print(int(menor))



if __name__ == '__main__':
    main()
